packages/server/lib/controllers/sync.controller.ts

```diff
@@ -77,11 +77,15 @@ class SyncController {
 
     private async addRecordCount(syncs: (Sync & { models: string[] })[], connectionId: number, environmentId: number) {
         const byModel = await recordsService.getRecordCountsByModel({ connectionId, environmentId });
-
         if (byModel.isOk()) {
             return syncs.map((sync) => ({
                 ...sync,
-                record_count: Object.fromEntries(sync.models.map((model) => [model, byModel.value[model]?.count ?? 0]))
+                record_count: Object.fromEntries(
+                    sync.models.map((model) => {
+                        const modelFullName = sync.variant === 'base' ? model : `${model}::${sync.variant}`;
+                        return [model, byModel.value[modelFullName]?.count ?? 0];
+                    })
+                )
```

Comment:
unrelated to the clearing cache issue but while testing I realized the records count on the connections page was showing the base sync count for all variants :(